Add initial support for new Trailblazer commodities and events

L00Cyph3r opened the pull request:
Did the following:
- Added the commodities as mentioned in #123
- Added support for `ColonisationContribution` and `ColonisationConstructionDepot` events
- Added my Journals for game-version: `4.1.0` (Trailblazers)

During testing my Journals, I did encounter one issue with `StationType`. I encountered `Market`-events which were legit, but the `StationType` was an empty string. Here are 2 examples:
Example with a Station that wasn't build yet at the time (iirc)
```json
{"timestamp":"2025-03-05T22:03:23Z","event":"Market","MarketID":4209655811,"StationName":"Nachtigal Platform","StationType":"","StarSystem":"Ceti Sector XJ-Z b2"}
```

And an example on one of the new Trailblazer ships (can be easily reproduced):
```json
{"timestamp":"2025-04-03T19:54:55Z","event":"Market","MarketID":129032439,"StationName":"Trailblazer Song","StationType":"","StarSystem":"Orgen"}
```

As far as I can see, there are 3 routes to take when it comes to the empty station-types.
1. Currently, there is an `Unknown(String)` for `StationType`, and that is what I am falling back on now. This does mean that the test without `--features allow-unknown` will fail.
2. Adding a `None` or something similar `StationType`, but this would require a custom serde-deserializer.
3. Changing the value for the higher-level stations from a `StationType` to an `Option<StationType>`. This would result in a breaking change though.

Since this is my first contribution, I'm going to leave this for input from others, as I don't feel comfortable yet with deciding which option would be best going forward.

rster2002 commented:
Hi! Looking really good already! Interesting to see the empty StationTypes. I'd probably want to go with adding a new entry `None` to the enum and making sure it's documented that it's none in cases where no type has been assigned (yet). If you want you can take a go at it or let me know then I'll take a look myself.

rster2002 commented on src/modules/logs/content/log_event_content/colonisation_construction_depot_event.rs:
```diff
@@ -0,0 +1,40 @@
+//! Fired when the player collected a commodity.
```
Left-over copy-pasted doc comment

rster2002 commented on src/modules/logs/content/log_event_content/colonisation_contribution_event.rs:
```diff
@@ -0,0 +1,32 @@
+use crate::trading::Commodity;
+use serde::{Deserialize, Serialize};
+
+/// Fired when the player collected a commodity.
```
Left-over copy-pasted doc comment

L00Cyph3r commented:
Changed the comments you mentioned.
Fixed the `StationType` issue.
Added a `None` version and used the `from_str_deserialize_impl!(StationType);` macro.
After fixing StationType, there were a few more additions with Trailblazers which I added as well.
All test succeed locally now.

_And on a sidenote, I cleaned my journals using your `ed-anonymize-journals`, and encountered an error because JavaScript can't handle 2^64-1 very well. This means that `"MissionID":18446744073709551615` gets messed up and results in `"MissionID":18446744073709552000`.
I fixed those manually, but this issue might return, so just a heads-up._

rster2002 commented:
Awesome work! Not surprised that JavaScript is messing things up, but for the test files it's fine. The ids are not something that is really something to test against anyways. Merged and creating a new release!
